Vc/detail/neon.h

```diff
@@ -528,6 +504,16 @@ struct neon_mask_impl {
     {
     }
     // masked store {{{2
+    template <class T, class F>
+    static Vc_INTRINSIC void masked_store(datapar<T> v, long double *mem, F,
+                                          mask<T> k) noexcept
+    {
+        execute_n_times<size<T>()>([&](auto i) {
+            if (k.d.m(i)) {
+                mem[i] = v.d.m(i);
+            }
+        });
+    }
```

Comment:
you re-added it in the mask_impl. That's wrong. And my question on the removal above was not meant to imply any resolution. It might actually make sense to remove the `long double` special-casing using some other mechanism to handle it like it were of type `double`.